src/base-traits.jl

```diff
@@ -1,15 +1,17 @@
 module BaseTraits
 using SimpleTraits
 
-export IsLeafType, IsBits, IsImmutable, IsContiguous, IsFastLinearIndex,
+using Compat
+
+export IsLeafType, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
        IsAnything, IsNothing, IsCallable
 
 "Trait which contains all types"
 @traitdef IsAnything{X}
 @traitimpl IsAnything{X} <- (x->true)(X)
 
 "Trait which contains no types"
-typealias IsNothing{X} Not{IsAnything{X}}
+@compat IsNothing{X} = Not{IsAnything{X}}
```

Comment:
the const is implicit for parametric type aliases, but IMO a useful signal about what the line is doing